doc/architecture/architecture.rst

```diff
@@ -1,24 +1,254 @@
+
+.. image:: images/ostro_os_architecture.svg
+   :width: 400px
+
+The following sections examine each layer in more details.
+
+IoT Applications
+----------------
+The IoT applications layer consist of all applications that use the
+underlying platform. Ostro OS itself currently does not
+implement any specific IoT use-cases via applications, but
+`application sample recipes`_ are provided to demonstrate the OS frameworks and services.
+
+Programming Interfaces
+-----------------------
+
+Ostro OS provides various application runtimes offering more flexibility and
+options to application developers to implement their IoT solution.
+
+* `Soletta Project`_ is an open-source
+  framework for making IoT applications. The project provides libraries
+  to make it quick and easy to write software for IoT devices. The
+  applications can we written using flow-based programming (FBP) or
+  more traditionally as C applications linking to Soletta's C-based
+  platform APIs.
+
+* `Node.JS`_ is a popular JavaScript runtime to
+  run (IoT) web applications. Ostro OS provides the Node.JS runtime
+  and selected JavaScript APIs (as Node.JS modules) to build IoT
+  applications. The set of APIs includes, for example, a Javascript API for
+  the `Open Connectivity Foundation`_ (OCF) specifications. The JavaScript
+  API set is kept aligned with the API set provided by Soletta.
+
+* Ostro OS also provides support for the popular IoT-space development
+  languages Python and Java\* (OpenJDK8).
+
+Frameworks
+----------
+
+Frameworks provide an abstraction to platform services, making
+application deployment easier. The Ostro Application
+Framework :ref:`application-framework` provides tooling for
+application writers to get the applications running in an Ostro OS
+based image.
+
+The Ostro Application framework implements a `systemd.generator`_ that
+parses the application manifest file to automatically generate a systemd service file.
+
+Ostro OS also pre-integrates `IoTivity`_ framework. IoTivity is an open-source
+project that implements the full OCF specification by `Open Connectivity Foundation`_. In
+addition to `IoTivity`_, another implementation of the specification API is also
+available via `Soletta Project`_.
+
+Services
+--------
+
+The responsibility of system services is to bring the system up,
+manage connectivity and set up process' inter-process communication (IPC).
+The components Ostro OS use are commonly found in many open-source OSes: systemd,
+ConnMan, BlueZ, D-Bus and others.
+
+In addition to systemd and connectivity, Ostro OS comes with
+:ref:`software-update` technology that helps support software
+updates to deployed devices. This technology is also used to replace traditional
+package management. This means Ostro OS comes with no RPMs or DEBs support. However,
+it can still be enabled if such feature is needed in a product/device.
+
+Ostro OS 1.0 focuses on headless use-cases and thus no graphics and
+multimedia are enabled by default.
```

Comment:
move this under "Base Libraries", you can rephrase as:

"While Ostro OS 1.0 is focused on headless use-cases and no graphic and multimedia libraries are enabled, one can use the Yocto Project to incorporate them to your product."